### PR TITLE
switch to rootlesskit from custom "unshare" binary

### PR DESCRIPTION
**Pull request by steved**
[Rootlesskit](github.com/rootless-containers/rootlesskit) seems to a better-supported, more feature-filled version of the custom [`unshare.c` we cribbed from `img`](https://github.com/genuinetools/img/blob/master/internal/unshare/unshare.c). In particular this should fix a hot-loop inside forge where a `Wait4` system call was executed continuously:
https://github.com/dominodatalab/forge/blob/238885c694f606e078f8c36e9ff308c56d35aabe/internal/buildjob/util.go#L114-L128

Inside rootlesskit:
https://github.com/rootless-containers/rootlesskit/blob/18143e8660646214bff613f6814400a22fafbfda/pkg/child/child.go#L306

### CODE REVIEW

File: controllers/containerimagebuild_resources.go [hunk 1]
```diff
@@ -286,6 +286,7 @@ func (r *ContainerImageBuildReconciler) jobForBuild(cib *forgev1alpha1.Container
 						{
 							Name:            "forge-build",
 							Image:           r.JobConfig.Image,
+							Command:         []string{"rootlesskit"},
```
Comment (sonnysideup) [inline controllers/containerimagebuild_resources.go:289]: const?

File: controllers/containerimagebuild_resources.go [hunk 2]
```diff
@@ -305,6 +306,7 @@ func (r *ContainerImageBuildReconciler) jobForBuild(cib *forgev1alpha1.Container
 // builds cli args required to launch forge in "build mode" inside a job
 func (r *ContainerImageBuildReconciler) prepareJobArgs(cib *forgev1alpha1.ContainerImageBuild) []string {
 	args := []string{
+		"/usr/bin/forge",
```
Comment (sonnysideup) [inline controllers/containerimagebuild_resources.go:309]: const?

File: Dockerfile [hunk 3]
```diff
@@ -41,6 +41,10 @@ RUN git clone -c advice.detachedHead=false https://github.com/opencontainers/run
 RUN make static BUILDTAGS="seccomp apparmor" && \
     cp runc /usr/bin/
 
+FROM gobase AS rootlesskit
+ENV GO111MODULE=on
```
Comment (sonnysideup) [inline Dockerfile:45]: Why set this envvar? I don't think you're in the GOPATH and there's no `go.mod`, right?

Read this cool article last week: https://dev.to/maelvls/why-is-go111module-everywhere-and-everything-about-go-modules-24k

Comment (steved) [reply Dockerfile:45]: Hmm
```
Step 18/40 : RUN go get github.com/rootless-containers/rootlesskit/cmd/rootlesskit@v0.10.0
 ---> Running in bda58396893f
go: cannot use path@version syntax in GOPATH mode
```

**Pull request merged**